docs/AUTH_PROTOCOL.md

```diff
@@ -0,0 +1,393 @@
+        - `overridePrompt`: Overrides the default prompt (or, more likely, adds a prompt) to the OTP entry UI. Optional. Should only be used if absolutely necessary
+    - Response: `response` event with arguments:
+        - `otp`: The OTP entered by the user. Neither the prefix nor the suffix are included.
+
+- `chooser`: UI for choosing from a list of options
+    - This method is always primary
+    - Arguments:
+        - `prompt`: A string to display to the user, explaining what they're choosing between. Optional.
+        - `options`: An array of objects that describes the options that the user can select between. Each object in the array has the following keys:
+            - `key`: The machine-usable string that identifies this option
+            - `name`: The display name for this option, to be presented to the user
+    - Response: `response` event with arguments:
+        - `selection`: The selected option's `key`
+
+- `eidp`: Web-based SSO login
+    - The frontend displays a QR code and/or an embedded web browser that loads a web page for single-sign-on
+    - This method is always primary
+    - Arguments:
+        - `url`: The URL of the SSO login page. The QR code encodes this URL, and the web browser navigates to the URL.
+        - `displayUrl`: A simplified URL to display to the user in text, suitible for the user to type manually. Optional
+        - `code`: A short code that is displayed to the user along with the `displayUrl`. This allows the user to complete authentication from an alternative device (like their phone)
+        - `showDone`: If true, the authenticator is unable to detect on its own when authentication has succeeded and the frontend should have a "Done" button for the user to press when they are done
+        - `embedded`: Allow the use of an embedded web view directly on the login screen. Defaults to true
+        - `external`: Allow the use of a QR code, for logging in via an external device (such as a phone). Defaults to true. 
+    - Response:
+        - if `showDone` is true, the frontend emits a `response` event (with no arguments) when the user clicks the done button
+        - if `showDone` is false, the frontend doesn't need to respond. The authenticator will continue with the conversation on its own
+
+- `fingerprint`: Fingerprint authentication
+    - The frontend prompts for the user to touch their fingerprint to the sensor, then listens for additional events (documented below) that the authenticator emits to communicate updates about fingerprint sensor state
+    - This method is either primary or secondary
+    - Arguments:
+        - `primary`: When true, this is a primary authentication mechanism
+        - `swipe`: When true, the fingerprint sensor requires a swiping motion of the finger, rather than a touch. Defaults to false
+    - Response: No response is necessary. The authenticator will continue with the conversation on its own
+
```

Comment:
Probably this does not fall into the `eidp` case, but another scenario that requires a super simple UI (basically just a message and a spinner or something) is:
 - Authenticate via magic-link
 - You receive a link, click on it from another device
 - You're in!

The UI should still show some message (_check your mailbox foo_), but not showing any QR code or anything else. Just a cancel button and maybe a "resend" button.

Reply:
OFC similar thing by receiving the magic link via SMS or for example if you have to just "confirm" an authentication in your mobile phone application, or verify a code there (like github app authenticator, while that google when asks you what number has been chosen is more a `pin` case)